test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -615,6 +618,80 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
 		)
 
 	})
+	// Automates OCPBUGS-34812: cgroupsv2: failed to write on cpuset.cpus.exclusive
+	Context("Cgroupsv2", func() {
+		It("[test_id:75327] cpus from deleted cgroup can be reassigned to new cgroup", Label(string(label.Tier0)), func() {
+
+			// we need system with more than 10 cpus to execute this test
+			if len(onlineCPUSet.List()) < 10 {
+				Skip("Requires system with  more than 10 cpus")
+			}
+			if !cgroupV2 {
+				Skip("Requires CgroupV2")
+			}
+			// create deployment with 2 replicas and each pod having cpu load balancing disabled
+			// and runtime class. This is required as the cpu id's used by the container are
+			// written to cpuset.cpus.exclusive
+			const DeploymentName = "test-deployment"
+			var numberofReplicas int32 = 2
+			var dp *appsv1.Deployment
+			annotations := map[string]string{
+				"cpu-load-balancing.crio.io": "disable",
+			}
+			p := pods.GetTestPod()
+			p.Spec.NodeSelector = testutils.NodeSelectorLabels
+			runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			p.Spec.RuntimeClassName = &runtimeClass
+			p.Spec.Containers[0].Image = images.Test()
+			p.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("200Mi"),
+					corev1.ResourceCPU:    resource.MustParse("2"),
+				},
+			}
+
+			// we create and delete deployment in loop to create deployments in
+			// quick succession to verify pre-start hook is able to write to
+			// cpuset.cpus.exclusive
+			for i := 0; i < 5; i++ {
+				// Create Deployment
+				dp = deployments.Make(DeploymentName, testutils.NamespaceTesting,
+					deployments.WithPodTemplate(p),
+					deployments.WithNodeSelector(testutils.NodeSelectorLabels))
+				dp.Spec.Template.Annotations = annotations
+				dp.Spec.Replicas = &numberofReplicas
+				Expect(testclient.DataPlaneClient.Create(ctx, dp)).ToNot(HaveOccurred())
+				podList := &corev1.PodList{}
+				listOptions := &client.ListOptions{Namespace: testutils.NamespaceTesting, LabelSelector: labels.SelectorFromSet(dp.Spec.Selector.MatchLabels)}
+				Eventually(func() bool {
+					if err := testclient.DataPlaneClient.List(context.TODO(), podList, listOptions); err != nil {
+						return false
+					}
+					//readyPods := 0
+					for _, pod := range podList.Items {
```

Comment:
if I understand correctly this eventually waits for the deployment to be ready, I think it would be a good idea to add that under a helper here maybe:
https://github.com/openshift/cluster-node-tuning-operator/tree/ea36e6ae28ce38898481d5f4cd8819a6e017c95b/test/e2e/util/wait
 also it is recommended to avoid nested if conditions as much as possible.

Reply:
It's bit difficult to avoid here, because there are just[ 5 phases](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) the pod can be  and in our case the pod though in oc status returns RunContainerError, its not a defined phase and this status comes only when the pod is in Pending state and crio is not able to write to cpuset.cpus.exclusive , so we need to check if the pod is in pending state and not all pending states give that error , hence we need to go again deeper to check waiting reason and if that reason is because pre-start hook failed. 

Even if we avoid the nested if's in test case, we have to the nested if's in some other utils function if we move code any where.